articles/execute_process_refactor.md

```diff
@@ -0,0 +1,596 @@
+
+#### Constructor
+
+These parameters are drawn from the shared parameters of the current `launch_ros.actions.Node` and `launch_ros.descriptions.ComposableNode`, and introduces a new collection of `traits`, described in following classes.
+
+| Argument       | Description                                                  |
+| -------------- | ------------------------------------------------------------ |
+| node_name      | Name the node should have                                    |
+| node_namespace | Namespace the node should create topics/services/etc in      |
+| parameters     | List of either paths to yaml files or dictionaries of parameters |
+| remappings     | List of from/to pairs for remapping names                    |
+| arguments      | Container specific arguments to be passed to the node        |
+| traits      | Additional features of the node, specifically Lifecycle or Composable information|
+
+#### Properties
+
+Accessors would be provided for the various constructor parameters: `node_name`, `node_namespace`, `parameters`, `remappings`, `arguments`, and `traits`.
+
+#### Methods
+
+In the current classes, substitutions are handled through internal method calls.
+To support better isolation of class functionality, substitutions will be performed in this class, but exposed from a public method which can be overridden by subclasses to perform additional appropriate substitutions.
+
+| Name    | Description                                                  |
+| ------- | ------------------------------------------------------------ |
+| prepare | Takes a given LaunchContext and Action, and performs actions necessary to prepare the node for execution in that context. This will primarily consist of expanding various substitutions, but may additionally add event handlers related to the Action which will be invoked. Additionally, this will call the `prepare()` method of any defined traits. |
+
+This function would match that of the current `launch_ros.actions.Node` internal method `_perform_substitutions`.
+
+#### Events
+
+No events would be handled or emitted.
+
+### launch_ros.descriptions.ComposableNode
+
+This class would extend the `launch_ros.descriptions.Node` class, and provide the additional information required for defining a node which can be launched in a composable context. 
+
+#### Constructor
+
+Most parameters would be passed to the superclass.
+
+|Argument|Description|
+|---|---|
+| package        | Optional. Name of the ROS package the node plugin lives in. If not specified, the package of the containing executable will be assumed. This may fail if context regarding the executable is not available.        |
+|node_plugin|Name of the plugin to be loaded|
+
+Additional parameters that may be passed, which are handled by `launch_ros.descriptions.NodeBase`: `node_name`, `node_namespace`, `parameters`, `remappings`, `arguments`, `traits`.
```

Comment:
I guess that `launch_ros.descriptions.NodeBase` is what was previously named `launch_ros.descriptions.Node`?

Reply:
Good catch, will update. Should be `launch_ros.descriptions.Node`; `NodeBase` was in a previous iteration, and has been removed.